app/src/main/java/com/sunasterisk/food_01/data/source/remote/api/ParseDataWithJson.kt

```diff
@@ -0,0 +1,68 @@
+package com.sunasterisk.food_01.data.source.remote.api
+
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlin.jvm.Throws
+
+class ParseDataWithJson {
+
+    @Throws(Exception::class)
+    fun getJsonFromUrl(urlString: String?): String? {
+        val url = URL(urlString)
+        val httpURLConnection = url.openConnection() as HttpURLConnection
+        httpURLConnection.apply {
+            connectTimeout = TIME_OUT
+            readTimeout = TIME_OUT
+            requestMethod = METHOD_GET
+            doOutput = true
+            connect()
+        }
+
+        val bufferedReader = BufferedReader(InputStreamReader(url.openStream()))
+        val stringBuilder = StringBuilder()
+        var line: String?
+        while (bufferedReader.readLine().also { line = it } != null) {
+            stringBuilder.append(line)
+        }
+        bufferedReader.close()
+        httpURLConnection.disconnect()
+        return stringBuilder.toString()
+    }
+
+    fun parseJsonToData(jsonObject: JSONObject?, keyEntity: String): Any {
+        val data = mutableListOf<Any>()
+        try {
+            val jsonArray = jsonObject?.getJSONArray(keyEntity)
+            for (i in 0 until (jsonArray?.length() ?: 0)) {
+                val jsonObjects = jsonArray?.getJSONObject(i)
+                val item = ParseDataWithJson().parseJsonToObject(jsonObjects, keyEntity)
+                item?.let { data.add(it) }
+            }
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        return data
+    }
+
+    private fun parseJsonToObject(jsonObject: JSONObject?, keyEntity: String): Any? {
+        try {
+            jsonObject?.let {
+                when (keyEntity) {
+                    else -> null
+                }
+            }
+        } catch (e: JSONException) {
+            e.printStackTrace()
+        }
+        return null
+    }
+
+    companion object {
+        private const val TIME_OUT = 15000
+        private val METHOD_GET = "GET"
```

Comment:
```suggestion
        private const val METHOD_GET = "GET"
```

**const val** và **val** khác nhau như nào nhỉ 
Ở đây dùng const val có được không ?

Reply:
**const val** là một top-level, bắt buộc phải có giá trị cố định
**val** nằm trong class, không cần có sẵn một giá trị cố định
ở đây theo em nghĩ là được